packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,391 @@
+      required: false,
+      default: () => ({start: '', end: ''})
+    },
+    timePeriods: {
+      type: Array,
+      required: false,
+      default: () => []
+    }
+  },
+
+  data () {
+    return {
+      hidePopover: false,
+      tabName: this.getDefaultTabName(),
+      abbreviatedDisplay: this.defaultMessage,
+      fullRangeDisplay: '',
+      selectedCalendarRange: this.defaultCustom,
+      selectedTimeframe: this.timePeriods[0],
+      selectedRange: { start: '', end: '' },
+      modelConfig: {
+        type: 'number'
+      },
+      calendarSelectAttributes: {
+        highlight: {
+          start: { class: 'vcal-day-start' },
+          base: { class: 'vcal-day-base' },
+          end: { class: 'vcal-day-end' },
+          day: { class: 'vcal-day-preselect' }
+        }
+      }
+    }
+  },
+
+  computed: {
+    showCalendar () {
+      return (this.tabName === 'custom')
+    },
+    hasRelativeTimeframes () {
+      return (this.timePeriods.length > 0)
+    },
+    submitDisabled () {
+      return (!this.selectedRange.start || !this.selectedRange.end)
+    },
+    hasDefaultCustomValue () {
+      return (this.defaultCustom.start && this.defaultCustom.end)
+    },
+    defaultTimeframe () {
+      return (this.defaultRelative !== undefined)
+        ? this.defaultRelative
+        : null
+    }
+  },
+
+  watch: {
+    // Updates input field's "human" date whenever v-calendar value is touched
+    selectedCalendarRange (newVal) {
+      if (newVal) {
+        const range = { start: this.msToSec(newVal.start), end: this.msToSec(newVal.end) }
+
+        this.abbreviatedDisplay = this.formatDisplayDate(range)
+      }
+    }
+  },
+
+  mounted () {
+    console.warn(' >>> defaultRelative: ')
+    console.log(this.defaultRelative)
+
+    // Select the tab based on incoming defaults
+    if (this.hasDefaultCustomValue) {
+      this.tabName = 'custom'
+    } else if (this.hasRelativeTimeframes) {
+      this.tabName = 'relative'
+    }
+
+    // Set default value to be displayed in the input field
+    if (this.hasDefaultCustomValue) {
+      const range = {
+        start: this.msToSec(this.defaultCustom.start),
+        end: this.msToSec(this.defaultCustom.end)
+      }
+
+      this.abbreviatedDisplay = this.formatDisplayDate(range)
+    } else if (this.defaultTimeframe) {
+      this.changeRelativeTimeframe(this.defaultTimeframe)
+    }
+  },
+
+  methods: {
+    /**
+     * Updates both the input field value, and the full time frame readout
+     * when a relative time frame button is clicked
+     * @param {*} timeframe
+     */
+    changeRelativeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+
+      // Update input field text
+      this.abbreviatedDisplay = this.selectedTimeframe.timeframeText
+
+      // Format the start/end values as human readable date
+      const end = getUnixTime(roundToNearestMinutes(Date.now()))
+      const start = end - this.selectedTimeframe.timeframeLength
```

Comment:
As per my original comment, this needs to be updated.
Cannot just do `now - timeframeLength` for all relative timeframes.

```
const end = this.selectedTimeframe.end()
const start = this.selectedTimeframe.start()
```